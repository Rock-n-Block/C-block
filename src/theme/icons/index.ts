export { ContractToken } from './components/ContractToken';
export { CrowdsaleIcon } from './components/CrowdsaleIcon';
export { KeyIcon } from './components/KeyIcon';
export { WeddingRingIcon } from './components/WeddingRingIcon';
export { WillContract } from './components/WillContract';
export { PlusCircleIcon } from './components/PlusCircleIcon';
export { FileTextIcon } from './components/FileTextIcon';
export { SettingGearIcon } from './components/SettingGearIcon';
export { AtIcon } from './components/AtIcon';
export { PlusIcon } from './components/PlusIcon';
export { PersonIcon } from './components/PersonIcon';
export { MoonIcon } from './components/MoonIcon';
export { CloseIcon } from './components/CloseIcon';
export { CopyIcon } from './components/CopyIcon';
export { EmailIcon } from './components/EmailIcon';
export { TrashIcon } from './components/TrashIcon';
export { ChevronIcon } from './components/ChevronIcon';
export { Menu } from './components/Menu';
