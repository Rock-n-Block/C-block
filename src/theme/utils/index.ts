import { Theme } from '@material-ui/core/styles';

export enum FormatMedia {
  BREAKPOINT_TABLET = 'BREAKPOINT_TABLET',
  BREAKPOINT_DESKTOP = 'BREAKPOINT_DESKTOP',
  BREAKPOINT_WIDE_DESKTOP = 'BREAKPOINT_WIDE_DESKTOP',
}

export const getFormatMedia = (theme: Theme): Record<FormatMedia, string> => ({
  BREAKPOINT_TABLET: theme.breakpoints.up('sm'),
  BREAKPOINT_DESKTOP: theme.breakpoints.up('md'),
  BREAKPOINT_WIDE_DESKTOP: theme.breakpoints.up('xl'),
});
