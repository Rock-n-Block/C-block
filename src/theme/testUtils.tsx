import React from 'react';

import { ThemeProvider as MuiThemeProvider } from '@material-ui/core';

import { theme } from './theme';

export const ThemeProvider: React.FC = ({ children }) => (
  <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
);
