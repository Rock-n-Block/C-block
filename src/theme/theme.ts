import { createTheme } from '@material-ui/core';
import { getMuiButton, getMuiButtonDefaultProps } from './Button';
import { COLOR_BLACK, COLOR_ERROR } from './colors';

import { getMuiContainer, getMuiContainerDefaultProps } from './Container';
import { getMuiCssBaseline, getMuiCssBaselineDefaultProps } from './CssBaseline';
import { getMuiDialogDefaultProps, getMuiDialogProps } from './Dialog';
import { getMuiGrid, breakpointOptions, getMuiGridDefaultProps } from './Grid';
import { getMuiIconButton } from './IconButton';
import { getMuiLinkDefaultProps, getMuiLinkOverride } from './Link';
import { getMuiSwitch, getMuiSwitchDefaultProps } from './Switch/Switch.theme';
import { getTypographyOptions } from './Typography';

// eslint-disable-next-line import/no-mutable-exports
export let theme = createTheme({
  palette: {
    error: {
      main: COLOR_ERROR,
    },
    primary: {
      main: COLOR_BLACK,
    },
  },
  typography: getTypographyOptions,
  breakpoints: breakpointOptions,
  spacing: 8,
});

theme = createTheme(theme, {
  props: {
    MuiCssBaseline: getMuiCssBaselineDefaultProps(),
    MuiLink: getMuiLinkDefaultProps(),
    MuiButtonBase: { disableRipple: true },
    MuiContainer: getMuiContainerDefaultProps(),
    MuiGrid: getMuiGridDefaultProps(),
    MuiTypography: getTypographyOptions(),
    MuiButton: getMuiButtonDefaultProps(),
    MuiSwitch: getMuiSwitchDefaultProps(),
    MuiDialog: getMuiDialogDefaultProps(),
  },
  overrides: {
    MuiCssBaseline: getMuiCssBaseline(theme),
    MuiLink: getMuiLinkOverride(theme),
    MuiContainer: getMuiContainer(theme),
    MuiGrid: getMuiGrid(theme),
    MuiButton: getMuiButton(theme),
    MuiIconButton: getMuiIconButton(theme),
    MuiSwitch: getMuiSwitch(theme),
    MuiDialog: getMuiDialogProps(theme),
  },
});
