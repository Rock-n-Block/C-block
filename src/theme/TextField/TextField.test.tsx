import React from 'react';

import { TextField } from '@material-ui/core';
import { render } from '@testing-library/react';

import { ThemeProvider } from 'theme/testUtils';

describe('TextField', () => {
  it('should render', () => {
    const { container } = render(
      <ThemeProvider>
        <TextField
          value="Value"
          placeholder="Placeholder"
          label="Label"
          helperText="Helper text"
          error
        />
      </ThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
