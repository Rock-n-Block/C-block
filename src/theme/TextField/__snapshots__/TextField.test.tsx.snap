// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should render 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root"
    data-testautomation-id="control"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error MuiFormLabel-filled"
      data-shrink="true"
    >
      Label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
    >
      <input
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input"
        placeholder="Placeholder"
        type="text"
        value="Value"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
        >
          <span>
            Label
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error MuiFormHelperText-filled"
    >
      Helper text
    </p>
  </div>
</div>
`;
